### ISSUE TITLE
Reproducing GRPO based on Qwen2.5-7B or DeepSeek-R1-distill-qwen7B and using NuminaMath-TIR dataset Yields Unexpected Results

### ISSUE THREAD

**Issue by GaryZhu1996** [huggingface/open-r1#358]
I am currently experimenting with the GPRO method in Open-R1, but the results have not met expectations.

I have tried **Qwen2.5-7B** and **Deeseek-R1-Distill-Qwen7B** as the base models and used **NuminaMath-TIR** as the training dataset. The experiments are conducted on 8xH100 with the following configuration:

```
bf16: true
use_vllm: true
vllm_device: auto
vllm_gpu_memory_utilization: 0.7
do_eval: true
eval_strategy: steps
eval_steps: 20
gradient_accumulation_steps: 16
gradient_checkpointing: true
gradient_checkpointing_kwargs:
  use_reentrant: false
learning_rate: 1.0e-05
log_level: info
logging_steps: 1
logging_strategy: steps
lr_scheduler_type: cosine
max_prompt_length: 512
max_completion_length: 2048
max_steps: -1
num_generations: 7
num_train_epochs: 2
save_steps: 50
overwrite_output_dir: true
per_device_eval_batch_size: 8
per_device_train_batch_size: 8
push_to_hub: false
report_to:
- wandb
save_strategy: "steps"
seed: 42
warmup_ratio: 0.01

# Rewards setting
reward_funcs: 
- accuracy
- format
- reasoning_steps
- cosine
```

Given this configuration, each epoch consists of approximately 500+ steps. However, I observed anomalies in the performance of checkpoints saved within the first 150 steps. For evaluation, I used AIME2024 and MATH-500 as test sets. However, as training progressed, the accuracy on the test sets steadily declined.

I am eager to determine the root cause of this issue. Could it be due to suboptimal hyperparameter settings? I would greatly appreciate any insights or suggestions you might have.

**Comment by Hasuer** [comment]
I had the same problem. I have done GRPO on Qwen2.5-1.5B-Instruct, as the base model, and use NuminaMath-TIR as the training dataset.
The performace no **gpqa:diamond** decline from 0.3081(base mode) to 0.2424 (RL model).

**Comment by GaryZhu1996** [comment]
Could you please let me know which dataset you used for training? Did you train for more than one epoch? Also, would you mind sharing your config settings? I noticed that although your performance did decrease, the drop in my experiment is much more significant than yours.

**Comment by Hasuer** [comment]
https://github.com/huggingface/open-r1/issues/363#issuecomment-2669237328
Here is the config I use.

**Comment by linkedlist771** [comment]
Hi, did you solve this? I also meet the same problem, the model's performance downgrade at 1000 steps checkpoints.
